### PR TITLE
Replace mirror-cn profile with Apache officially suggested dlcdn

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Replace `mirror-cn` profile with Apache officially suggested dlcdn.

mirror-cn is only suite for CN, while dlcdn is much wiser.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by ulysses-you — APPROVED** [review]
failed test is not related.

```
- check discovery service is clean up with different share level *** FAILED ***
  5,5,1645637582680,1645637582680,0,2,0,0,9,0,8
   did not equal null (ShareLevelSparkEngineSuite.scala:53)
```

File: pom.xml [hunk 1]
```diff
@@ -1810,10 +1811,10 @@
 
     <profiles>
         <profile>
-            <id>mirror-cn</id>
+            <id>mirror-cdn</id>
+            <!-- this profile works only for lastest apache releases -->
```
Comment (pan3793) [inline pom.xml:1815]: typo: lastest

File: docs/develop_tools/building.md [hunk 2]
```diff
@@ -73,22 +73,18 @@ Profile | Default  | Since
...
 
-Visit [Apache Mirrors](http://www.apache.org/mirrors/) and choose a mirror based on your region.
-
-Specifically for developers in China mainland, you can use the pre-defined profile named `mirror-cn` 
- which use `mirrors.bfsu.edu.cn` to speed up Spark Binary downloading. For instance,
-
-```bash
-build/mvn clean package -Pmirror-cn
-```
+The profile migrates your download repo to the Apache offically suggested site - https://dlcdn.apache.org.
```
Comment (pan3793) [inline docs/develop_tools/building.md:88]: Do we need to quote the "https://dlcdn.apache.org" by `

Comment (yaooqinn) [reply docs/develop_tools/building.md:88]: ` is not for links

**Pull request closed without merging**